pkg/aggregator/context_resolver.go

```diff
@@ -231,8 +255,8 @@ func newTimestampContextResolver(tagger tagger.Component, cache *tags.Store, id
 }
 
 // trackContext returns the contextKey associated with the context of the metricSample and tracks that context
-func (cr *timestampContextResolver) trackContext(metricSampleContext metrics.MetricSampleContext, currentTimestamp int64) ckey.ContextKey {
-	contextKey := cr.resolver.trackContext(metricSampleContext, currentTimestamp)
+func (cr *timestampContextResolver) trackContext(metricSampleContext metrics.MetricSampleContext, currentTimestamp int64, filterList *utilstrings.Matcher) ckey.ContextKey {
+	contextKey := cr.resolver.trackContext(metricSampleContext, currentTimestamp, filterList)
```

Comment:
Would it be possible to pass the filter list (or a dynamic container for it) during construction?

Reply:
Nvm, I think this is actually better way (no need to synchronize for updates) even though it makes the api a bit messier.